Sources/Analytics/Analytics.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  Analytics.swift
+//  Alicerce
+//
+//  Created by Luís Portela on 26/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+/// Analytics allows you to track pages or events to all the registered trackers.
+/// Registered trackers should implement the `AnalyticsTracker` protocol.
+/// For every track, the parameters sent with the `Page` or `Event` are merged with global parameters.
+public final class Analytics {
+    
+    /// Typealias for a dictionary with `String` as key and with `Any` value
+    public typealias Parameters = [String : Any]
+    
+    public var parameters: Parameters? { return extraParameters }
+    
+    fileprivate lazy var trackers = [AnalyticsTracker]()
+    
+    fileprivate let trackingQueue: DispatchQueue
+    
+    fileprivate var extraParameters: Parameters?
+    
+    /// Creates an analytics instance with provided configuration
+    ///
+    /// - Parameter configuration: A Configuration object with QoS and extraParameters (if any)
+    init(configuration: Configuration = Configuration()) {
+        self.trackingQueue = DispatchQueue(label: "com.mindera.alicerce.queue.analytics.tracking",
+                                           qos: configuration.queueQoS)
+        self.extraParameters = configuration.extraParameters
+    }
+    
+    // MARK: - Public Methods
+    
+    /// Appends a new parameter into the extraParameters.
+    /// Those are appended with tracking item when a track is called.
+    ///
+    /// - Parameter parameters: A dictionary with a String and Any
+    public func add(parameters: Parameters) {
+        if let _ = extraParameters {
```

Comment:
if we are unwrapping here, we can use the unwrapped value and avoid optional chaining 😛